common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawDataBank.java

```diff
@@ -0,0 +1,107 @@
+package org.jlab.detector.banks;
+
+import org.jlab.jnp.hipo4.data.Bank;
+import org.jlab.jnp.hipo4.data.Event;
+import org.jlab.jnp.hipo4.data.Schema;
+import org.jlab.jnp.hipo4.io.HipoReader;
+
+/**
+ * FilterBank specific to raw ADC/TDC banks, filtered on order by decades.  This
+ * is to leverage hijacking the higher digits of order to encode additional
+ * information, and avoid reconstruction engines or others from manually
+ * interpreting order.  See RawOrderType for conventions.
+ * 
+ * @author baltzell
+ */
+public class RawDataBank extends FilteredBank {
+
+    public RawDataBank(Schema sch, int allocate){
+        super(sch, allocate, "order");
+    }
+
+    public final void setFilter(RawOrderType... types) {
+        for (int i=0; i<types.length; i++) {
+            for (int j = 0; j<10; j++) {
+                filterList.add(j + types[i].getTypeId());
+            }
+        }
+    }
+
+    /**
+     * Maybe we should name this something else?
+     * @param index filtered index
+     * @return raw/true order, the first digit
+     */
+    public int trueOrder(int index){
+        return this.intValue("order", index)%10;
+    }
+
+    /**
+     * @param index filtered index
+     * @return sector value
+     */
+    public int sector(int index){
+        return this.intValue("sector", index);
+    }
+    
+    /**
+     * @param index filtered index
+     * @return layer value
+     */
+    public int layer(int index){
+        return this.intValue("layer", index);
+    }
+    
+    /**
+     * @param index filtered index
+     * @return component value
+     */
+    public int component(int index){
+        return this.intValue("component", index);
+    }
+    
+    /**
+     * @param index filtered index
+     * @return adc value
+     */
+    public int adc(int index){
```

Comment:
adc banks have additional variables, I assume we will need methods for those as well?

Reply:
no